spec/controllers/invite_requests_controller_spec.rb

```diff
@@ -179,16 +179,48 @@
     end
 
     context "when logged in as admin" do
+      let(:ip) { "127.0.0.1" }
+      let(:ip_2) { "128.0.0.1" }
       let!(:invite_request_1) { create(:invite_request, position: 9001) }
       let!(:invite_request_2) { create(:invite_request, position: 2) }
       let!(:invite_request_3) { create(:invite_request, position: 7) }
+      let!(:invite_request_4) do
+        create(
+          :invite_request,
+          position: 500,
+          email: "hello_world@gmail.com"
+        )
+      end
 
-      before { fake_login_admin(admin) }
+      before do
+        fake_login_admin(admin)
+      end
+
+      it "searches invitations by email" do
+        get :manage, params: { query: "hello_world" }
+        expect(response).to render_template("manage")
+        expect(assigns(:invite_requests)).to eq([invite_request_4])
+      end
+
+      it "searches invitations by ip" do
+        invite_request_4.ip_address = ip
```

Comment:
Thanks! Just one more minor thing: you can set the IPs in the `create` calls at the start instead of re-saving them here.
```rb
let!(:invite_request_1) { create(:invite_request, position: 9001, ip_address: "128.0.0.1") }
```